### PR TITLE
Add futures 0.3 and use async/await in some places

### PR DESCRIPTION
**Pull request by jameswestman**
Add futures 0.3 so we can use async/await. For now, futures 0.1 and 0.3 are both used, and actix-web is not updated to 4.0 yet.

### CODE REVIEW

File: src/api.rs [hunk 1]
```diff
@@ -214,41 +225,53 @@ pub fn create_build(
...
 }
 
 pub fn builds(
     db: Data<Db>,
     req: HttpRequest,
 ) -> impl Future<Item = HttpResponse, Error = ApiError> {
-    futures::done(req.has_token_claims("build", "build"))
-        .and_then(move |_| db.list_builds())
-        .and_then(move |builds| Ok(HttpResponse::Ok().json(builds)))
+    Box::pin(builds_async(db, req)).compat()
```
Comment (bilelmoussaoui) [inline src/api.rs:268]: This should probably be marked as async so you no longer need to pin it.

Comment (bilelmoussaoui) [reply src/api.rs:268]: I also wonder what is the difference between build & builds_async here?

Comment (jameswestman) [reply src/api.rs:268]: As I understand it, actix-web 1.0 uses futures 0.1, which doesn't support async/await. I haven't figured out how to upgrade everything to actix-web 4.0 yet, so for now this converts the async/await futures to 0.1 futures for actix-web.

I'm not a rust expert, so I'm sure there's a better way to do this somehow.

Comment (bilelmoussaoui) [reply src/api.rs:268]: Ah yes, my bad. I missed the compat part somehow.

**Pull request merged**